### PR TITLE
[PLAY-54]: added network-graph component

### PR DESCRIPTION
**Pull request by AntonNos**
Added `network-graph` component as a own npm package.
Screenshot 2023-05-25 at 21 28 42

### CODE REVIEW

File: src/.DS_Store
Comment (Christopher-Gustavsson) [inline src/.DS_Store:1]: Lets add `.DS_Store` to the `.gitignore` file so we avoid this file.

File: src/index.test.tsx
Comment (Christopher-Gustavsson) [inline src/index.test.tsx:1]: We can probably remove this test. I'm guessing this was just to see if the configs we working or something like that.

**Review by Christopher-Gustavsson — CHANGES_REQUESTED** [review]
Awesome work. Just left some minor comments nothing big. 
Other than those I think this is looking pretty good. Good idea to include an example!

File: src/.DS_Store
Comment (AntonNos) [reply src/.DS_Store:1]: Oh, sorry, I forgot about this one.
Removed and added it ti `.gitignore`

File: src/index.test.tsx
Comment (AntonNos) [reply src/index.test.tsx:1]: Yes, right. I added it to check that config works well. Removed

**Comment by AntonNos** [comment]
> Awesome work. Just left some minor comments nothing big. Other than those I think this is looking pretty good. Good idea to include an example!

Thanks! Comments were fixed.

File: src/index.test.tsx
Comment (AntonNos) [reply src/index.test.tsx:1]: Hmm, I pushed update without test and build fails. Probably we need to add at least 1 test

Comment (AntonNos) [reply src/index.test.tsx:1]: I added another test file, which includes simple test for navigation controls

File: package.json [hunk 1]
```diff
@@ -1,49 +1,98 @@
...
+    "@typescript-eslint/eslint-plugin": "^5.59.7",
+    "@typescript-eslint/parser": "^5.59.7",
+    "@visx/responsive": "^3.0.0",
+    "@visx/zoom": "^3.1.1",
+    "babel-core": "7.0.0-bridge.0",
+    "babel-eslint": "^10.0.3",
+    "cross-env": "^7.0.2",
     "css-loader": "^6.7.3",
-    "eslint": "^8.39.0",
+    "d3-drag": "^3.0.0",
```
Comment (dechegaray) [inline package.json:53]: let's move these d3-packages to `dependencies`

File: package.json [hunk 2]
```diff
@@ -1,49 +1,98 @@
...
-    "@types/react-dom": "^18.2.1",
-    "@types/testing-library__jest-dom": "^5.14.5",
-    "@typescript-eslint/eslint-plugin": "^5.59.2",
-    "@typescript-eslint/parser": "^5.59.2",
+    "@types/node": "^20.2.3",
+    "@types/react": "^18.2.7",
+    "@types/react-dom": "^18.2.4",
+    "@typescript-eslint/eslint-plugin": "^5.59.7",
+    "@typescript-eslint/parser": "^5.59.7",
+    "@visx/responsive": "^3.0.0",
```
Comment (dechegaray) [inline package.json:47]: visx/responsive and zoom should also go to `dependencies`

**Review by dechegaray — CHANGES_REQUESTED** [review]
A couple of important changes we need to have the package as clean as possible 👍

File: dist/index.js
[the remark this answers is not in the document]
Comment (AntonNos) [reply dist/index.js:1]: Done

File: example/src/App.tsx
[the remark this answers is not in the document]
Comment (AntonNos) [reply example/src/App.tsx:1]: Done

File: package.json [hunk 3]
```diff
@@ -1,49 +1,98 @@
...
     "format": "prettier --write \"{src,tests,example/src}/**/*.{js,ts,jsx,tsx}\"",
+    "deploy": "gh-pages -d example/build",
     "typecheck": "tsc --noEmit --declaration false"
   },
   "dependencies": {
+    "babel-jest": "^29.5.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-scripts": "^5.0.1",
     "typescript": "^5.0.4"
```
[the remark this answers is not in the document]
Comment (AntonNos) [reply package.json:32]: Done

File: package.json [hunk 4]
```diff
@@ -1,49 +1,98 @@
...
+    "prettier": "^2.0.4",
     "react-test-renderer": "^18.2.0",
     "style-loader": "^3.3.2",
     "ts-jest": "^29.1.0",
     "ts-loader": "^9.4.2",
-    "type-check": "^0.4.0",
-    "webpack": "^5.81.0",
-    "webpack-cli": "^5.0.2",
-    "webpack-dev-server": "^4.13.3"
+    "type-check": "^0.4.0"
```
[the remark this answers is not in the document]
Comment (AntonNos) [reply package.json:81]: It is used for github action - `"typecheck": "tsc --noEmit --declaration false"`

File: package.json [hunk 5]
```diff
@@ -1,49 +1,98 @@
...
     "file-loader": "^6.2.0",
-    "html-webpack-plugin": "^5.5.1",
+    "gh-pages": "^5.0.0",
     "jest": "^29.5.0",
     "jest-canvas-mock": "^2.5.0",
     "jest-environment-jsdom": "^29.5.0",
     "jest-junit": "^16.0.0",
-    "prettier": "^2.8.8",
+    "microbundle-crl": "^0.13.10",
+    "npm-run-all": "^4.1.5",
```
[the remark this answers is not in the document]
Comment (AntonNos) [reply package.json:75]: I removed this one - `"npm-run-all": "^4.1.5"`, but `microbundle-crl` is used for `start` and `build` scripts

File: package.json [hunk 6]
```diff
@@ -1,49 +1,98 @@
...
+    "eslint-plugin-react": "^7.17.0",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-standard": "^5.0.0",
     "file-loader": "^6.2.0",
-    "html-webpack-plugin": "^5.5.1",
+    "gh-pages": "^5.0.0",
     "jest": "^29.5.0",
     "jest-canvas-mock": "^2.5.0",
     "jest-environment-jsdom": "^29.5.0",
     "jest-junit": "^16.0.0",
```
[the remark this answers is not in the document]
Comment (AntonNos) [reply package.json:73]: Done

File: README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
+# network-graph
+
+> Made with create-react-library
+
+[![NPM](https://img.shields.io/npm/v/network-graph.svg)](https://www.npmjs.com/package/network-graph) [![JavaScript Style Guide](https://img.shields.io/badge/code_style-standard-brightgreen.svg)](https://standardjs.com)
+
+## Install
+
+```bash
+npm install --save network-graph
```
[the remark this answers is not in the document]
Comment (AntonNos) [reply README.md:10]: Done

File: package.json (hunk 1 above)
Comment (AntonNos) [reply package.json:53]: Done

File: package.json (hunk 2 above)
Comment (AntonNos) [reply package.json:47]: Done

**Comment by AntonNos** [comment]
> A couple of important changes we need to have the package as clean as possible 👍

Thanks for comments, I double checked packages and removed unused. Other comments were also fixed.

**Review by dechegaray — APPROVED** [review]

**Pull request merged**